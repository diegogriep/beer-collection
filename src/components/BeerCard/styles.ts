import styled, { css } from 'styled-components'
import media from 'styled-media-query'

export const Wrapper = styled.div`
  ${({ theme }) => css`
    background-color: ${theme.colors.primary};
    border-radius: ${theme.border.radius};
    color: #000;
    padding: ${theme.spacings.small};
    width: 100%;

    h1 {
      font-size: ${theme.font.sizes.large};
      text-transform: uppercase;
    }

    h2 {
      font-size: ${theme.font.sizes.small};
    }

    div {
      column-gap: ${theme.spacings.xsmall};
      display: flex;
      padding: ${theme.spacings.xsmall} 0;
    }

    p {
      font-size: ${theme.font.sizes.small};
    }

    a {
      font-size: ${theme.font.sizes.small};
    }

    ${media.greaterThan('small')`
      width: 29rem;
    `}
  `}
`
