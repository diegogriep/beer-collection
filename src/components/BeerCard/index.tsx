import Image from 'next/image'
import Button from '../Button'
import * as S from './styles'

export type IngredientsBeerProps = {
  name: string
  amount: IngredientAmountProps
  add?: string
  attribute?: string
}

export type IngredientsKindProps = {
  [name: string]: IngredientsBeerProps[]
}

type IngredientAmountProps = {
  value: number
  unit: string
}

export type BeerProps = {
  id: number
  name: string
  description: string
  image_url: string
  tagline: string
  ingredients: IngredientsKindProps
  brewers_tips: string
}

const BeerCard = ({
  id,
  name,
  image_url,
  tagline,
  brewers_tips
}: Partial<BeerProps>) => (
  <S.Wrapper>
    <h1>{name}</h1>
    <h2>{tagline}</h2>

    <div>
      <Image
        priority
        style={{ width: 75, height: 75 }}
        alt={name!}
        src={image_url!}
        width={75}
        height={128}
      />

      <p>{brewers_tips}</p>
    </div>

    <Button as="a" href={`/beer/${id}`} title="Learn more">
      Learn more
    </Button>
  </S.Wrapper>
)

export default BeerCard
