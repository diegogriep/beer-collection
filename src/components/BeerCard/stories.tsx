import { Meta, StoryObj } from '@storybook/react'
import BeerCard, { BeerProps } from '.'

import beer from './mock'

export default {
  title: 'components/BeerCard',
  component: BeerCard,
  args: { ...beer[0] }
} as Meta

export const Default: StoryObj<BeerProps> = {
  render: (args) => (
    <BeerCard {...args} />
  )
}
