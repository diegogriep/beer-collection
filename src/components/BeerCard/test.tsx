import { screen } from '@testing-library/react'

import BeerCard from '.'
import { renderWithTheme } from '../../utils/tests/helpers'

import beer from './mock'

describe('<BeerCard />', () => {
  it('should render the heading', () => {
    const { container } = renderWithTheme(<BeerCard {...beer} />)

    expect(screen.getByRole('heading', { name: /Buzz/i })).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
