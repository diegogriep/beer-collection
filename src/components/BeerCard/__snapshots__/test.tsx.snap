// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BeerCard /> should render the heading 1`] = `
.c1 {
  background: linear-gradient(180deg, #C96E12 0%, #DF8D03 50%);
  border: 0;
  border-radius: 0.4rem;
  color: #fff;
  display: inline-flex;
  cursor: pointer;
  font: 700 1.6rem mocked,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,sans-serif;
  padding: 0.8rem 1.6rem;
  text-decoration: none;
}

.c1:hover {
  background: linear-gradient(180deg, #DF8D03 0%, #C96E12 50%);
}

.c0 {
  background-color: #FAE96F;
  border-radius: 0.4rem;
  color: #000;
  padding: 2.4rem;
  width: 100%;
}

.c0 h1 {
  font-size: 1.8rem;
  text-transform: uppercase;
}

.c0 h2 {
  font-size: 1.4rem;
}

.c0 div {
  column-gap: 1.6rem;
  display: flex;
  padding: 1.6rem 0;
}

.c0 p {
  font-size: 1.4rem;
}

.c0 a {
  font-size: 1.4rem;
}

@media (min-width: 450px) {
  .c0 {
    width: 29rem;
  }
}

<div
  class="c0"
>
  <h1>
    Buzz
  </h1>
  <h2>
    A Real Bitter Experience.
  </h2>
  <div>
    <img
      alt="Buzz"
      data-nimg="1"
      decoding="async"
      height="128"
      loading="lazy"
      src="/_next/image?url=https%3A%2F%2Fimages.punkapi.com%2Fv2%2Fkeg.png&w=256&q=75"
      srcset="/_next/image?url=https%3A%2F%2Fimages.punkapi.com%2Fv2%2Fkeg.png&w=96&q=75 1x, /_next/image?url=https%3A%2F%2Fimages.punkapi.com%2Fv2%2Fkeg.png&w=256&q=75 2x"
      style="color: transparent;"
      width="75"
    />
    <p>
      The earthy and floral aromas from the hops can be overpowering. Drop a little Cascade in at the end of the boil to lift the profile with a bit of citrus.
    </p>
  </div>
  <a
    class="c1"
    href="/beer/1"
    title="Learn more"
  >
    Learn more
  </a>
</div>
`;
